backend/src/v5/middleware/dataConverter/inputs/auth.js

```diff
@@ -0,0 +1,39 @@
+/**
+ *  Copyright (C) 2021 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+const { createResponseCode, templates } = require('../../../utils/responseCodes');
+const Yup = require('yup');
+const { respond } = require('../../../utils/responder');
+
+const Auth = {};
+
+Auth.validateLoginData = async (req, res, next) => {
+	const schema = Yup.object().shape({
+		username: Yup.string().required(),
+		password: Yup.string().required(),
+	}).strict(true).noUnknown()
+		.required();
+
+	try {
+		await schema.validate(req.body);
```

Comment:
once it's validated, can we convert email address to user name here? then we don't have to worry about if it's a username or email address in the processor.

If the user is not found, we could actually just return here